04-using-github-with-the-platform.Rmd

```diff
@@ -192,6 +192,62 @@ The platform has configured simple "safety barriers" to reduce risk of accidenta
 | Large files (>5 Mb) | ~/.git-templates/hooks/pre-commit | Likely to be data | When you commit: `git commit --no-verify` |
 | Pushing to non-official GitHub organizations | ~/.git-templates/hooks/pre-push | It would be outside MoJ control - not normally allowed. | When you push: `git push -f <remote> <branch>` |
 
+## Private R Packages on GitHub: PAT Authentication
+
+### Private and Public Repositories
+
+The majority of repos in the DASD GitHub organisation are **Private**: this keeps visibility, access and editing rights controlled to particular groups within the MoJ network and is sensible in most cases. Owners and Admins of a repository can control this visibility by going to **_Settings_ > _Collaborators & Teams_** and adding teams to read/write/admin access groups. To maximise collaboration _within_ the organsiation but still keep code that shouldn't be in the public domain (e.g. unpublished commentary on a `.Rmd` file) hidden from anyone _outside_ the organisation, you can add the group `everyone` here. This also makes your code [searchable.](https://moj-analytical-services.github.io/platform_user_guidance/using-github-with-the-platform.html#other-tips-and-tricks-work-in-progress)
+
+### Private R Packages for Reproducible Analysis
+
+For the most part, this private group-based access works well and seamlessly. 
+Where it can rub is in building **R Packages** for high-quality reproducable analysis either across small teams or across the organisation.
+The AP team have built a number of _generic_ tools that are non-sensitive in nature (e.g. [`s3tools`](github.com/moj-analytical-services/s3tools) and [`dbtools`](github.com/moj-analytical-services/dbtools)) and as such are in fact **Public** repos that anyone, inside or outside the organisation, can view, access, and install to their library.
+If an internal package is being used for a specific project, RAP or other analytical work, the chances are that it will not be appropriate to make these codebases public, as some combination of the structure, code and comments can be revealing, however [sensibly the data are stored](https://moj-analytical-services.github.io/platform_user_guidance/working-with-big-datasets-and-sharing-them-accessing-data-in-amazon-s3.html).
```

Comment:
Definitely agree with this, particularly for packages for general RAP functions, because they shouldn't contain information about the data to which they are being applied.